includes/admin/js/geolocator.js

```diff
@@ -1,53 +1,194 @@
+				region = feature.text;
+			}
+
+		// Otherwise, we have a full result set, so let's parse its data
+		} else {
+			for ( var i = 0; i < feature.context.length; i++ ) {
+				if ( feature.context[i].id.match( /^place/ ) ) {
+					city = feature.context[i].text;
+				}
+				if ( feature.context[i].id.match( /^country/ ) ) {
+					country = feature.context[i].text;
 				}
-				var lng = response.features[0].center[0];
-				var lat = response.features[0].center[1];
-				$( '#wanderlist-city' ).val( city );
-				$( '#wanderlist-region' ).val( region );
-				$( '#wanderlist-country' ).val( country );
-				$( '#wanderlist-lng' ).val( lng );
-				$( '#wanderlist-lat' ).val( lat );
-				var placeName = response.features[0].place_name;
-
-				// Vatican doesn't return the correct results, so we're going to do it manually
-				if ( 'Città del Vaticano' === city ) {
-					$( '#wanderlist-region' ).val( '' );
-					$( '#wanderlist-country' ).val( 'Vatican City' );
-					placeName = 'Città del Vaticano';
+				if ( feature.context[i].id.match( /^region/ ) ) {
+					region = feature.context[i].text;
 				}
+			}
 
-				$( '#wanderlist-geocoder-message' ).addClass( 'success' );
-				$( '#wanderlist-geocoder-message' ).find( '.place' ).text( placeName );
-			},
-			error: function(response) {
-				$( '#wanderlist-geocoder-message' ).addClass( 'error' );
-				$( '#wanderlist-geocoder-message' ).find( '.place' ).text( 'ERROR' );
+			// If the city hasn't already been set, we assume that we can get it via the feature name
+			if ( ! city ) {
+				city = feature.text;
+			}
+		}
+
+		// Set lat and long
+		var lng = feature.center[0];
+		var lat = feature.center[1];
+
+		// Output our values to our hidden form fields
+		$( '#wanderlist-city' ).val( city );
+		$( '#wanderlist-region' ).val( region );
+		$( '#wanderlist-country' ).val( country );
+		$( '#wanderlist-lng' ).val( lng );
+		$( '#wanderlist-lat' ).val( lat );
+		var placeName = feature.place_name;
+
+		// Since Geocoder results for Vatican aren't accurate, we'll overwrite them
+		if ( 'Città del Vaticano' === city ) {
+			$( '#wanderlist-region' ).val( '' );
+			$( '#wanderlist-country' ).val( 'Vatican City' );
+			placeName = 'Città del Vaticano';
+		}
+
+		// Show a friendly success message and show point on map
+		$( '#wanderlist-geocoder-message' ).removeClass( 'error' );
+		$( '#wanderlist-geocoder-message' ).addClass( 'success' );
+		$( '#wanderlist-geocoder-message' ).find( '.place' ).text( placeName );
+		showPointOnMap( lat, lng );
+	}
+
+	/*
+	 * Get the user's current location.
+	 * This uses the HTML5 geolocation API, available on
+ 	 * most mobile browsers and modern browsers
+	 * http://caniuse.com/#feat=geolocation
+	 */
+	 function getCurrentLocation( button ) {
+		 // If our browser doesn't support geolocation, hide the button for a more graceful degredation
+		 if ( ! navigator.geolocation ) {
+			 button.innerHTML = 'Geolocation is not available';
```

Comment:
Why not use the jQuery API? (`$(button).html('Geolocation is not available');`)

I forget if it uses `innerHTML` underneath but explicit `innerHTML` usage I recall being bad... but then maybe that's no longer the case?